main_test.go

```diff
@@ -239,6 +250,89 @@ func TestCreateScenario(t *testing.T) {
 	}
 }
 
+func TestCreateScenarioTLS(t *testing.T) {
+	// prepare TLS files
+	cert, certKey := generateCerts()
+	certFile, keyFile, err := createCertPairFiles(cert, certKey)
+	if err != nil {
+		t.Fatalf("Failed to prepare certs %v", err)
+	}
+	defer os.Remove(certFile.Name())
+	defer os.Remove(keyFile.Name())
+
+	gen := types.ScenarioItem{}
+	err = gen.ParseTLS(certFile.Name(), keyFile.Name())
+	if err != nil {
+		t.Fatalf("Failed to gen certs %v", err)
+	}
+
+	url := "https://test.com"
+	valid := types.Scenario{
+		Scenario: []types.ScenarioItem{
+			{
+				ID:       1,
+				Protocol: types.DefaultProtocol,
+				Method:   types.DefaultMethod,
+				URL:      url,
+				Timeout:  types.DefaultDuration,
+				Cert:     gen.Cert,
+				CertPool: gen.CertPool,
+			},
+		},
+	}
+
+	certPathArg := fmt.Sprintf("--cert_path=%s", certFile.Name())
+	keyPathArg := fmt.Sprintf("--cert_key_path=%s", keyFile.Name())
+
+	tests := []struct {
+		name      string
+		args      []string
+		shouldErr bool
+		expected  types.Scenario
+	}{
+		{"MissingKey", []string{"-t=https://test.com", certPathArg}, false, types.Scenario{}},
+		{"MissingCert", []string{"-t=https://test.com", keyPathArg}, false, types.Scenario{}},
+		{"WithTLS", []string{"-t=https://test.com", certPathArg, keyPathArg}, false, valid},
+	}
+
+	for _, test := range tests {
+		tf := func(t *testing.T) {
+			// Arrange
+			resetFlags()
+			oldArgs := os.Args
+			defer func() {
+				os.Args = oldArgs
+			}()
+
+			os.Args = []string{"cmd"}
+			for _, a := range test.args {
+				os.Args = append(os.Args, a)
+			}
+
+			// Act
+			flag.Parse()
+			s, err := createScenario()
+
+			// Assert
+			if test.shouldErr {
+				if err == nil {
+					t.Errorf("Should be errored")
+				}
+			} else {
+				if err != nil {
+					t.Errorf("Errored: %v", err)
+				}
+				if reflect.DeepEqual(test.expected, s) {
```

Comment:
this should be `!reflect.DeepEqual(test.expected, s)`.
Looks like same bug happened on  `TestCreateScenario`. I'll send a fix for it on another PR.

Reply:
Here is the fixed version of `TestCreateScenario`.  If you would like, you can send this fix along with this PR.

```
func TestCreateScenario(t *testing.T) {
	url := "https://test.com"
	valid := types.Scenario{
		Scenario: []types.ScenarioItem{
			{
				ID:       1,
				Protocol: types.DefaultProtocol,
				Method:   types.DefaultMethod,
				URL:      url,
				Timeout:  types.DefaultTimeout,
				Headers:  map[string]string{},
			},
		},
	}
	validWithAuth := types.Scenario{
		Scenario: []types.ScenarioItem{
			{
				ID:       1,
				Protocol: types.DefaultProtocol,
				Method:   types.DefaultMethod,
				URL:      url,
				Timeout:  types.DefaultTimeout,
				Headers:  map[string]string{},
				Auth: types.Auth{
					Type:     types.AuthHttpBasic,
					Username: "testuser",
					Password: "pass",
				},
			},
		},
	}

	tests := []struct {
		name      string
		args      []string
		shouldErr bool
		expected  types.Scenario
	}{
		{"InvalidAuth", []string{"-t=https://test.com", "-a=no_pass_included"}, true, types.Scenario{}},
		{"InvalidTarget", []string{"-t=asds.x.x.x"}, true, types.Scenario{}},
		{"Valid", []string{"-t=https://test.com"}, false, valid},
		{"ValidWithAuth", []string{"-t=https://test.com", "-a=testuser:pass"}, false, validWithAuth},
	}

	for _, test := range tests {
		tf := func(t *testing.T) {
			// Arrange
			resetFlags()
			oldArgs := os.Args
			defer func() {
				os.Args = oldArgs
			}()

			os.Args = []string{"cmd"}
			for _, a := range test.args {
				os.Args = append(os.Args, a)
			}

			// Act
			flag.Parse()
			s, err := createScenario()

			// Assert
			if test.shouldErr {
				if err == nil {
					t.Errorf("Should be errored")
				}
			} else {
				if err != nil {
					t.Errorf("Errored: %v", err)
				}
				if !reflect.DeepEqual(test.expected, s) {
					t.Errorf("Expected %#v, Found %#v", test.expected, s)
				}
			}

		}

		t.Run(test.name, tf)
	}
}
```

Reply:
I sent the fix

Reply:
@zhiburt I think this issue still exists. I fixed the `TestCreateScenario`. `TestCreateScenarioTLS` still has incorrect comparison.

Reply:
fixed